### PR TITLE
Feature/csv export activation

### PR DESCRIPTION
**Pull request by mvaivre**
Seems to be working on the FE side. 

One thing that I don't like: this ugly way to start the download from a blob (ie. anchor a temp link, then programatically clicking on it... what is it, 1999?). I quickly checked around, didn't find any obvious more modern solution.

### CODE REVIEW

File: src/modals/ExportAddressTXsModal.tsx [hunk 1]
```diff
@@ -16,44 +16,107 @@ You should have received a copy of the GNU Lesser General Public License
...
+
+      if (!blob) throw 'Problem while downloading the CSV file'
+
+      const url = URL.createObjectURL(blob)
+      const a = document.createElement('a')
+      a.style.display = 'none'
+      a.href = url
+      a.download = `${addressHash}__${timePeriodValue}__${dayjs().format('DD-MM-YYYY')}` // Filename
+      document.body.appendChild(a)
+      a.click()
```
[the remark this answers is not in the document]
Comment (nop33) [reply src/modals/ExportAddressTXsModal.tsx:77]: What we could do a bit more in the "React way" would be to define an element in the render function, use state to populate it with data, and use `useRef` to click it programmatically.

```tsx
const ref = useRef()
...
ref.current.click()
...
<TemporaryDownloadLink ref={ref} style={{ display: 'none' }} href={url} download={download} />
```

[the remark this answers is not in the document]
Comment (mvaivre) [reply src/modals/ExportAddressTXsModal.tsx:77]: I'm not sure I like this more. At least with the current solution we don't pollute the vitual dom => what we see in the render part of the component is actually what we get. 

I could maybe extract the current logic out of the component, to clean it up a bit.

File: src/components/Select.tsx [hunk 2]
```diff
@@ -141,42 +141,88 @@ const Select = ({
...
+interface SelectItemProps<T> {
+  value: T
+  label?: string
+  LabelComponent?: ReactNode
+  title?: string
+  isDrawerOpen: boolean
+  isSelected?: boolean
+  onClick: (value: T) => void
+  initialItemHeight: Required<Dimensions>['initialItemHeight']
+  expandedItemHeight: Required<Dimensions>['expandedItemHeight']
```
[the remark this answers is not in the document]
Comment (mvaivre) [reply src/components/Select.tsx:162]: It's actually `number | undefined`

image

In this subcomponent, I want those value to be set (and I know that they are as the parent set default values for them).

File: src/components/Select.tsx [hunk 3]
```diff
@@ -141,42 +141,88 @@ const Select = ({
...
+      height: expandedItemHeight,
+      transition
+    }
+  }
+
+  return (
+    <ItemContainer
+      key={value}
+      onClick={() => !isSelected && onClick(value)}
+      className={isSelected ? 'selected' : undefined}
```
[the remark this answers is not in the document]
Comment (nop33) [reply src/components/Select.tsx:194]: OK, I see your point. Happy to discuss and experiment.

> I find that having many ${({ something, else}) => something && else ...} in styled components make them hard to read as it's a lot of boilerplate, especially when we have many conditions in there.

I agree. In other places we use a similar pattern, which I find a bit cleaner:

```ts
${({ initialItemHeight, alignText }) => css`
  height: ${initialItemHeight};
  justify-content: ${alignText};
`}
```

My only concern about having dynamic styles in the render function is that we are mixing styles with JSX code, resulting in a bloated render function expanding in many lines. I personally like seeing a compact render function with as little CSS/JS code as possible. But, of course, there are always exceptions, like simple callback functions like:

```ts
onClick={(e) => handleClick(e.target.value)}
```

or using the main component's props to derive child component styles through the child's `style` prop.

File: src/modals/ExportAddressTXsModal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nop33) [reply src/modals/ExportAddressTXsModal.tsx:77]: > I could maybe extract the current logic out of the component, to clean it up a bit.

sounds great

**Comment by mvaivre** [comment]
Man that was painful. The logic when managing errors in `swagger-typescript-api` is super weird.

**Pull request merged**